src/crimson/common/tri_mutex.h

```diff
@@ -27,22 +27,15 @@ class excl_lock {
 // promote from read to excl
 class excl_lock_from_read {
 public:
-  seastar::future<> lock();
-  void unlock();
+  void promote();
+  void demote();
```

Comment:
this change is confusing. the naming of `lock` and `unlock` follows the convention of the lock concept in standard library. and it reflect the semantics of the "lock" in `excl_lock_from_read`. i would vote to keep the name, if i could.

Reply:
I don't have strong preference here. I'm good to change back to the names lock/unlock.

Reply:
But I still want to differentiate `lock() -> future<>` vs `lock() -> void`. How about rename to `lock_sync()` or `lock_excl()` ?

Reply:
Renamed back to lock() and protect with concepts instead.